### PR TITLE
Improvements for docker mode

### PR DESCRIPTION
**Pull request by alexsaveliev**
- removed hack for srclib-java in Docker mode
- returning error instead of panicking when we were unable to normalize repo clone URI

### CODE REVIEW

File: cli/test_cmd.go [hunk 1]
```diff
@@ -114,6 +114,16 @@ func (c *TestCmd) Execute(args []string) error {
 			}
 			expectedDir := filepath.Join(tree, "../../expected", exeMethod, filepath.Base(tree))
 			actualDir := filepath.Join(tree, "../../actual", exeMethod, filepath.Base(tree))
+
+			_, err := os.Stat(expectedDir)
+			if err != nil && os.IsNotExist(err) {
```
Comment (beyang) [inline cli/test_cmd.go]: `if _, err := os.Stat(expectedDir); err != nil && os.IsNotExist(err) {`

File: cli/test_cmd.go [hunk 2]
```diff
@@ -114,6 +114,16 @@ func (c *TestCmd) Execute(args []string) error {
...
+				// (alexsaveliev) we may have some tests available only for docker or program mode
```
Comment (beyang) [inline cli/test_cmd.go]: Why would a test only exist for docker or program mode and not both? The two should have equivalent behavior (as much as practically possible).

File: cli/test_cmd.go (hunk 1 above)
Comment (alexsaveliev) [inline cli/test_cmd.go]: fixed

File: cli/test_cmd.go (hunk 2 above)
Comment (alexsaveliev) [inline cli/test_cmd.go]: Right, they should behave identical as much as possible, but what Docker image has some pre-requisites missing (e.g. Android SDK)?

File: cli/test_cmd.go [hunk 3]
```diff
@@ -114,6 +114,15 @@ func (c *TestCmd) Execute(args []string) error {
 			}
 			expectedDir := filepath.Join(tree, "../../expected", exeMethod, filepath.Base(tree))
 			actualDir := filepath.Join(tree, "../../actual", exeMethod, filepath.Base(tree))
+
+			if _, err := os.Stat(expectedDir); err != nil && os.IsNotExist(err) {
```
Comment (beyang) [inline cli/test_cmd.go]: `if _, err := os.Stat(expectedDir); os.IsNotExist(err) {`

Comment (beyang) [inline cli/test_cmd.go]: Actually, can we just remove this clause, and generate expected output for both the program and Docker cases in srclib-java?

**Comment by beyang** [comment]
after comment is addressed, LGTM.

File: cli/test_cmd.go (hunk 3 above)
Comment (alexsaveliev) [inline cli/test_cmd.go]: Reverted changes related to abliity to exclude missing tests

**Pull request merged**